osu.Game/Overlays/BeatmapListing/BeatmapSearchMultipleSelectionFilterRow.cs

```diff
@@ -73,7 +73,12 @@ protected override void LoadComplete()
             private void currentChanged(object? sender, NotifyCollectionChangedEventArgs e)
             {
                 foreach (var c in Children)
-                    c.Active.Value = Current.Contains(c.Value);
+                {
+                    if (!c.Active.Disabled)
+                        c.Active.Value = Current.Contains(c.Value);
+                    else if (c.Active.Value != Current.Contains(c.Value))
+                        throw new InvalidOperationException($"Expected filter {c.Value} to be set to {Current.Contains(c.Value)}, but was {c.Active.Value}");
```

Comment:
How can this happen?

Reply:
If some code decides to remove the featured artists filter externally via `filter.General.Current.Remove(FeaturedArtists)`, it will trigger this path.

What is being sent to the API is the value of `filter.General.Current`, so if the state detaches like that it's more correct to be loud about it as in the line above rather than silently failing. Your call.

Reply:
All I know is that I can't understand these two lines of code in isolation, so it probably needs a comment or just removing. I don't see it being necessary as part of this change. Let's remove for now.

Reply:
Sure, I've only done it as a general tactic of ensuring data integrity seeing that there are multiple bindables conveying the state of the "featured artists" tab. I guess I'll save my tactics for a PR that actually needs it 😅

Reply:
https://github.com/ppy/osu/pull/31238/commits/7e3477f4bbfaa9cb1c01dea68b320e7267c5bbda